GCC-Toolchain have problem in scalapack compilation

### Describe the bug

When using `toolchain_gcc.sh`  by default, scalapack build will fail by :
```
make[2]: *** [TESTING/LIN/CMakeFiles/xzls.dir/build.make:207: TESTING/xzls] Error 1
make[2]: Leaving directory '/data/test_admin/abacus-develop/toolchain/build/scalapack-2.2.1/build'
make[1]: *** [CMakeFiles/Makefile2:2277: TESTING/LIN/CMakeFiles/xzls.dir/all] Error 2
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `ompi_mpi_op_sum'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Bcast'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Op_create'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Pack_size'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Send'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `ompi_mpi_packed'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Isend'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Unpack'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `ompi_mpi_comm_world'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Initialized'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Type_free'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Barrier'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Group_incl'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Sendrecv'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Finalize'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Rsend'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Recv'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Testall'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Comm_group'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Comm_create'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `ompi_mpi_double'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `ompi_mpi_float'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Pack'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `ompi_mpi_cplex'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Abort'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Type_create_struct'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Comm_size'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Waitall'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Allreduce'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `ompi_mpi_comm_null'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Type_commit'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Wtime'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Reduce'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Comm_dup'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Comm_f2c'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Irecv'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Error_class'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Comm_get_attr'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Comm_rank'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Init'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `ompi_mpi_unsigned_short'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Op_free'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `ompi_mpi_dblcplex'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Type_match_size'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Comm_free'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Comm_split'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Type_vector'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Type_indexed'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Group_free'
/usr/bin/ld: ../../lib/libscalapack.so.2.2.1: undefined reference to `MPI_Comm_c2f'
collect2: error: ld returned 1 exit status
make[2]: *** [TESTING/EIG/CMakeFiles/xssep.dir/build.make:342: TESTING/xssep] Error 1
```

### Expected behavior

build complete

### To Reproduce

_No response_

### Environment

_No response_

### Additional Context

I'll try to fix it by downgrading the default openmpi version.

### Task list for Issue attackers (only for developers)

- [ ] Verify the issue is not a duplicate.
- [ ] Describe the bug.
- [ ] Steps to reproduce.
- [ ] Expected behavior.
- [ ] Error message.
- [ ] Environment details.
- [ ] Additional context.
- [ ] Assign a priority level (low, medium, high, urgent).
- [ ] Assign the issue to a team member.
- [ ] Label the issue with relevant tags.
- [ ] Identify possible related issues.
- [ ] Create a unit test or automated test to reproduce the bug (if applicable).
- [ ] Fix the bug.
- [ ] Test the fix.
- [ ] Update documentation (if necessary).
- [ ] Close the issue and inform the reporter (if applicable).

--- a/toolchain/scripts/stage3/install_scalapack.sh
+++ b/toolchain/scripts/stage3/install_scalapack.sh
@@ -59,8 +59,7 @@ case "$with_scalapack" in
         -DCMAKE_VERBOSE_MAKEFILE=ON \
         -DBUILD_SHARED_LIBS=YES \
         -DCMAKE_BUILD_TYPE=Release .. \
-        -DCMAKE_C_FLAGS:STRING="$CFLAGS -Wno-implicit-function-declaration"
-        -DBUILD_TESTING=NO \
+        -DCMAKE_C_FLAGS:STRING="$CFLAGS -Wno-implicit-function-declaration" \
         -DSCALAPACK_BUILD_TESTS=NO \
         > configure.log 2>&1 || tail -n ${LOG_LINES} configure.log
       make -j $(get_nprocs) > make.log 2>&1 || tail -n ${LOG_LINES} make.log

